<?xml version="1.0"?>

<launch>
	<arg name="UNIT" default="Celsius"/>
	<arg name="ACCESS_URI" default="http://localhost:8442/serviceregistry/"/>
	<arg name="THIS_ADDRESS" default="130.240.5.68"/>
	<arg name="SERVICE_URI" default="Custom_url"/>
	<arg name="THIS_SYSTEM_NAME" default="temperature_provider_demo"/>
	<arg name="SERVICE_NAME" default="temperature_c"/>
	<arg name="INTERFACE" default="JSON"/>
	<arg name="SECURITY" default="Token"/>
	<arg name="THIS_PORT" default="8490"/>
	<arg name="SECURE_ARROWHEAD_INTERFACE" default="false"/>
	<arg name="SECURE_PROVIDER_INTERFACE" default="false"/>

	<arg name="OR_URI" default="http://localhost:8440/orchestrator/orchestration"/>
	<arg name="SERVICE_NAME_CONSUMER" default="terget name"/>
	<arg name="TARGET_SYSTEM_NAME" default="target name"/>
	<arg name="TARGET_ADDRESS" default="ip to target"/>
	<arg name="TARGET_PORT" default="1234"/>
	<arg name="OVERRIDE_STORE" default="true"/>
	<arg name="MATCHMAKING" default="true"/>
	<arg name="METADATA_SEARCH" default="false"/>
	<arg name="PING_PROVIDERS" default="false"/>
	<arg name="ONLY_PREFERRED" default="true"/>
	<arg name="EXTERNAL_SERVICE_REQUEST" default="false"/>
	<arg name="REAL" default="false"/>
	
	<!-- service -->
	<node pkg="provider" type="provider" name="provider">
		<param name="PRIVATE_KEY_PATH" value="Absolute_path_to_private_key"/>
		<param name="PUBLIC_KEY_PATH" value="Absolute_path_to_public_key"/>
		<param name="ACCESS_URI" value="$(arg ACCESS_URI)"/>
		<param name="THIS_ADDRESS" value="$(arg THIS_ADDRESS)"/>
		<param name="SERVICE_URI" value="$(arg SERVICE_URI)"/>
		<param name="THIS_SYSTEM_NAME" value="$(arg THIS_SYSTEM_NAME)"/>
		<param name="SERVICE_NAME" value="$(arg SERVICE_NAME)"/>
		<param name="INTERFACE" value="$(arg INTERFACE)"/>
		<param name="UNIT" value="$(arg UNIT)"/>
		<param name="SECURITY" value="$(arg SECURITY)"/>
		<param name="THIS_PORT" value="$(arg THIS_PORT)"/>
		<param name="SECURE_ARROWHEAD_INTERFACE" value="$(arg SECURE_ARROWHEAD_INTERFACE)"/>
		<param name="SECURE_PROVIDER_INTERFACE" value="$(arg SECURE_PROVIDER_INTERFACE)"/>

		<param name="OR_URI" value="$(arg OR_URI)"/>
		<param name="SERVICE_NAME_CONSUMER" value="$(arg SERVICE_NAME_CONSUMER)"/>
		<param name="TARGET_SYSTEM_NAME" value="$(arg TARGET_SYSTEM_NAME)"/>
		<param name="TARGET_ADDRESS" value="$(arg TARGET_ADDRESS)"/>
		<param name="TARGET_PORT" value="$(arg TARGET_PORT)"/>
		<param name="OVERRIDE_STORE" value="$(arg OVERRIDE_STORE)"/>
		<param name="MATCHMAKING" value="$(arg MATCHMAKING)"/>
		<param name="METADATA_SEARCH" value="$(arg METADATA_SEARCH)"/>
		<param name="PING_PROVIDERS" value="$(arg PING_PROVIDERS)"/>
		<param name="ONLY_PREFERRED" value="$(arg ONLY_PREFERRED)"/>
		<param name="EXTERNAL_SERVICE_REQUEST" value="$(arg EXTERNAL_SERVICE_REQUEST)"/>
		<param name="REAL" value="$(arg REAL)"/>
	</node>


</launch>

